src/json.cpp

```diff
@@ -550,6 +677,301 @@ bool Json::Node::is_map() const
   return YAJL_IS_OBJECT(JSON_TREE_HANDLE);
 }
 
+bool add_value(yajl_val obj, const std::string &key, yajl_val value)
+{
+    const char **temp_key = NULL;
+    yajl_val *temp_value = NULL;
+    char *key_copy = NULL;
+    
+    temp_key = (const char **) realloc((void*) obj->u.object.keys,
+                                       sizeof(*(obj->u.object.keys)) * (obj->u.object.len + 1));
+    
+    if (temp_key == NULL) {
+        return false;
+    }
+    obj->u.object.keys = temp_key;
+    
+    temp_value = (yajl_val *) realloc((void*) obj->u.object.values,
+                                      sizeof(*(obj->u.object.values)) * (obj->u.object.len + 1));
+    
+    if (temp_value == NULL) {
+        return false;
+    }
+    
+    obj->u.object.values = temp_value;
+    
+    key_copy = (char*) malloc(key.length() + 1);
+    memcpy(key_copy, key.c_str(), key.length());
+    key_copy[key.length()] = '\0';
+    
+    if (key_copy == NULL)
+    {
+        return false;
+    }
+
+    obj->u.object.keys[obj->u.object.len] = key_copy;
+    obj->u.object.values[obj->u.object.len] = value;
+    obj->u.object.len++;
+    
+    return true;
+}
+
+bool add_value(yajl_val obj, yajl_val value)
+{
+    yajl_val *temp_value = NULL;
+    
+    temp_value = (yajl_val *) realloc((void *) obj->u.array.values,
+                                      sizeof(*(obj->u.array.values)) * (obj->u.array.len + 1));
+    
+    if (temp_value == NULL) {
+        return false;
+    }
+    
+    obj->u.array.values = temp_value;
+    obj->u.array.values[obj->u.array.len] = value;
+    obj->u.array.len++;
+    
+    return true;
+}
+
+yajl_val add_number(yajl_val obj, const std::string &key, int int_value, double double_value, int flags)
+{
+    yajl_val value = new_number(int_value, double_value, flags);
+    
+    if (value != NULL)
+    {
+        if (!add_value(obj, key, value))
+        {
+            free(value->u.number.r);
+            free(value);
+            return NULL;
+        }
+    }
+        
+    return value;
+}
+
+yajl_val add_number(yajl_val obj, int int_value, double double_value, int flags)
+{
+    yajl_val value = new_number(int_value, double_value, flags);
+    
+    if (value != NULL)
```

Comment:
Sounds like a use case for Template Functions, Windows should be able to handle them by now.